testsuite/cluster-test/src/experiments/compatibility_test.rs

```diff
@@ -0,0 +1,187 @@
+        let mut first_batch = first_batch.into_validator_instances();
+        let first_node = first_batch
+            .pop()
+            .expect("Requires at least one validator in the first batch");
+
+        Self::E {
+            first_node,
+            first_batch,
+            second_batch: second_batch.into_validator_instances(),
+            full_nodes: cluster.fullnode_instances().to_vec(),
+            updated_image_tag: self.updated_image_tag,
+        }
+    }
+}
+
+#[async_trait]
+impl Experiment for CompatibilityTest {
+    fn affected_validators(&self) -> HashSet<String> {
+        instance::instancelist_to_set(&self.first_batch)
+            .union(&instance::instancelist_to_set(&self.second_batch))
+            .cloned()
+            .chain(once(self.first_node.peer_name().clone()))
+            .collect()
+    }
+
+    async fn run(&mut self, context: &mut Context<'_>) -> anyhow::Result<()> {
+        let validator_txn_job = EmitJobRequest::for_instances(
+            context.cluster.validator_instances().to_vec(),
+            context.global_emit_job_request,
+        );
+        let fullnode_txn_job = EmitJobRequest::for_instances(
+            context.cluster.fullnode_instances().to_vec(),
+            context.global_emit_job_request,
+        );
+        let job_duration = Duration::from_secs(3);
+
+        // Generate some traffic
+        context
+            .tx_emitter
+            .emit_txn_for(job_duration, fullnode_txn_job.clone())
+            .await?;
+
+        info!("1. Changing the images for the first instance to validate storage");
+        let first_node = vec![self.first_node.clone()];
+        update_batch_instance(context, &first_node, self.updated_image_tag.clone()).await?;
+        context
+            .tx_emitter
+            .emit_txn_for(
+                job_duration,
+                EmitJobRequest::for_instances(first_node, context.global_emit_job_request),
+            )
+            .await?;
+
+        info!("2. Changing images for the first batch to test consensus");
+        update_batch_instance(context, &self.first_batch, self.updated_image_tag.clone()).await?;
+        context
+            .tx_emitter
+            .emit_txn_for(job_duration, validator_txn_job.clone())
+            .await?;
+
+        info!("3. Changing images for the rest of the nodes");
+        update_batch_instance(context, &self.second_batch, self.updated_image_tag.clone()).await?;
+        context
+            .tx_emitter
+            .emit_txn_for(job_duration, validator_txn_job)
+            .await?;
+
+        info!("4. Changing images for full nodes");
+        update_batch_instance(context, &self.full_nodes, self.updated_image_tag.clone()).await?;
+        context
+            .tx_emitter
+            .emit_txn_for(job_duration, fullnode_txn_job)
+            .await?;
+        Ok(())
+    }
+
+    fn deadline(&self) -> Duration {
+        Duration::from_secs(20 * 60)
```

Comment:
How long does it normally take? Guideline is to put here 2 * normal_experiment_duration.
Too large values aren't great as sometimes it may have experiment running when it simply hanged

Reply:
Each restart will take 2min at most. There were four restarts. So I would imagine the execution time would be 4*2 min + txn generation. Given such calculation 20min seems fine? We might want to reduce it to 16min just to constraint the time consumption a little bit.

Reply:
Make sense!